newsfeed-ionic/src/app/services/newsfeed-data.service.ts

```diff
@@ -42,16 +43,11 @@ export class NewsfeedDataService {
     this.newsfeed.splice(index, 1);
   }
 
-  findPost(id: number): IPost {
-    const index: number = this.newsfeed.findIndex((post) => post.id === id);
-    return this.newsfeed[index];
-  }
-
-  addReaction(postId: number, userId: number, typeOfReaction: ReactionTypes = ReactionTypes.Like) {
+  addReaction(postId: number, typeOfReaction: ReactionType) {
     const post: IPost = this.findPost(postId);
     const reactions: IReaction[] = post.reactions;
     const reactionIndex: number = reactions.findIndex(
-      (reactionValue) => reactionValue.reactedBy.id === userId && reactionValue.reactionType === typeOfReaction,
+      (reactionValue) => reactionValue.reactedBy.id === this.defaultUser.id && reactionValue.reactionType === typeOfReaction,
     );
 
     const foundReaction: IReaction = post.reactions[reactionIndex];
```

Comment:
Why not just simplify this to [find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead?